player/Controls.gd

```diff
@@ -23,108 +24,112 @@ var _is_swimming_down: bool = false
-    else:
-        # otherwise we just make the mobile controls node visible
-        _mobile_controls.visible = true
+	# wait until the parent node is ready
+	await get_parent().ready
+
+	# OS.has_touchscreen_ui_hint() reports true for anything with touch support including HTML5
+	# which isn't exactly what we want when it comes down to enabling touchscreen controls
+	# so instead we check if the OS name is Android or iOS, and if so, we enable touch controls
+	var os_name := OS.get_name()
+	if os_name == "Android" || os_name == "iOS":
+		_is_touchscreen = true
+
+	# if touchscreen controls are disabled we capture the mouse in the game window
+	if !_is_touchscreen:
+		_is_capturing = true
+		Input.set_mouse_mode(Input.MOUSE_MODE_CAPTURED)
+	else:
+		# otherwise we just make the mobile controls node visible
+		_mobile_controls.visible = true
 
 func _process(delta):
-    # if the mouse cursor is being captured in the game window
-    if _is_capturing:
-        # determine the movement direction based checked the input strengths of the four movement directions
-        var dx := Input.get_action_strength("move_right") - Input.get_action_strength("move_left")
-        var dy := Input.get_action_strength("move_forward") - Input.get_action_strength("move_backwards")
-
-        # and set the movement direction vector to the normalized vector so the player can't unintentionally
-        # move faster when moving diagonally
-        _move_vec = Vector2(dx, -dy).normalized()
-    elif _is_touchscreen:
-        # othwerise if we're checked a touchscreen device, get the movement direction, camera rotation and
-        # zoom scale values from the mobile controls node
-        _move_vec = _mobile_controls.get_movement_vector()
-        _cam_rot = _mobile_controls.get_camera_rotation()
-        _zoom_scale = _mobile_controls.get_zoom_scale()
-
-    # in both desktop and touch screen devices the jump flag can be determined via the jump action
-    # same goes for other actions
-    _is_jumping = Input.is_action_just_pressed("jump")
-    _is_sprinting = Input.is_action_pressed("sprint")
-    _is_dashing = Input.is_action_pressed("dash")
-    _is_crouching = Input.is_action_pressed("crouch")
-    _is_swimming_up = Input.is_action_pressed("swim_up")
-    _is_swimming_down = Input.is_action_pressed("swim_down")
-    _is_surging = Input.is_action_pressed("surge")
+	# if the mouse cursor is being captured in the game window
+	if _is_capturing:
+		# determine the movement direction based checked the input strengths of the four movement directions
+		var dx := Input.get_action_strength("move_right") - Input.get_action_strength("move_left")
+		var dy := Input.get_action_strength("move_forward") - Input.get_action_strength("move_backwards")
+		# rotate the camera with controller
+		_cam_rot -= Input.get_vector("pan_left", "pan_right", "pan_up", "pan_down") * controller_sensitivity
+		_cam_rot.y = clamp(_cam_rot.y, min_pitch, max_pitch)
```

Comment:
Does this interfere with the mouse controls? Can we still use the mouse without an issue even if a controller is plugged in or not?